### DISCUSSION TITLE
Yolov8 Pyinstaller exe file constantly reruns itself

### DISCUSSION THREAD

**Discussion by alierenaltindag** [ultralytics/ultralytics#4153] [Q&A] [answered]
I converted a very simple application that I developed with Yolov8 to exe with pyinstaller. The script runs very well but when I run the exe file it constantly reruns itself. My codes are as follows:
```js
from ultralytics import YOLO

model = YOLO("best.pt")
results = model.track("./last.mp4", show=True)
```
Pyinstaller command to convert it to exe file:
```
pyinstaller main.py --no-console
```
Errors:
![image](https://github.com/ultralytics/ultralytics/assets/47426589/c1ec5c37-d9b2-4c9f-b16a-e0e9a09c6ea9)

Additional note:
After converting the script to an exe file, I manually created the ultralytics/cfg/default.yaml file at the location of the exe file because it gave an error that ultralytics could not find the default.yaml file.

**Comment by alierenaltindag** [comment, marked as answer]
I discovered that the issue is related to the 'cpuinfo' module. I don't yet know the reason, but removing the relevant code from the Ultralytics codebase resolved the problem.

**Reply by glenn-jocher** [reply]
@alierenaltindag it's great to hear that you've identified the issue with the 'cpuinfo' module and found a workaround. It's possible that the 'cpuinfo' module may not be compatible with the PyInstaller environment or there could be a path resolution issue when the executable is run.

For a more permanent solution, you might consider exploring the PyInstaller hooks or the spec file to ensure that all necessary files and modules are correctly included during the conversion process. Additionally, you can check if there are any known issues with 'cpuinfo' and PyInstaller compatibility and if there are any updates or patches available.

If you continue to encounter issues or have further questions about using YOLOv8, please refer to our documentation at https://docs.ultralytics.com for guidance on various modes and tasks supported by YOLOv8. If the problem persists, feel free to open an issue on the GitHub repository with detailed information, and we'll be happy to assist you further. 🚀
